server/src/main/java/org/opentosca/toscana/plugins/cloudformation/CloudFormationFileCreator.java

```diff
@@ -82,10 +84,25 @@ private void createDeployScript() throws IOException {
         logger.debug("Creating deploy script.");
         BashScript deployScript = new BashScript(cfnModule.getFileAccess(), FILENAME_DEPLOY);
         deployScript.append(EnvironmentCheck.checkEnvironment("aws"));
+        deployScript.append("source " + FILENAME_UPLOAD + ".sh");
         deployScript.append(CHANGE_TO_PARENT_DIRECTORY);
-        deployScript.append(CLI_COMMAND_CREATESTACK
-            + CLI_PARAM_STACKNAME + cfnModule.getStackName() + " "
-            + CLI_PARAM_TEMPLATEFILE + TEMPLATE_YAML);
+        StringBuilder deployCommand = new StringBuilder("");
+        deployCommand.append(CLI_COMMAND_CREATESTACK + CLI_PARAM_STACKNAME)
+            .append(cfnModule.getStackName()).append(" ")
+            .append(CLI_PARAM_TEMPLATEFILE).append(TEMPLATE_YAML);
+
+        // Add parameters if needed
+        Map<String, Parameter> parameters = cfnModule.getParameters();
+        if (!CollectionUtils.isEmpty(parameters)) {
```

Comment:
Whats the difference to `!parameters.isEmpty()` ?

Reply:
This also provides safety for null checks, so if the collection is null, there is no NullPointerException. That cannot happen here (I think) so `!parameters.isEmpty()` would, as you suggested, suffice.

I just personally like using CollectionUtils.isEmpty() in general, so I don't run into unexpected NullPointerExceptions.

Reply:
Often though, crashing early on encountering a nullpointer is better than propagating the error even further...